src/main/java/de/tum/p2p/api/netty/forwarding/OnionTunnelIncomingMessage.java

```diff
@@ -0,0 +1,45 @@
+package de.tum.p2p.api.netty.forwarding;
+
+import de.tum.p2p.onion.forwarding.TunnelId;
+import de.tum.p2p.proto.message.MessageType;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+
+import java.nio.ByteBuffer;
+import java.security.KeyFactory;
+import java.security.PublicKey;
+import java.security.spec.X509EncodedKeySpec;
+
+@ToString
+@EqualsAndHashCode(callSuper = true)
+public class OnionTunnelIncomingMessage extends OnionApiMessage {
+    private final TunnelId tunnelId;
+    private final PublicKey publicKey;
+
+    public OnionTunnelIncomingMessage(TunnelId tunnelId, PublicKey publicKey) {
+        super(MessageType.ONION_TUNNEL_READY);
+        this.tunnelId = tunnelId;
+        this.publicKey = publicKey;
+    }
+
+    public static OnionTunnelIncomingMessage fromBytes(ByteBuffer buffer) throws Exception {
+
+        int id = buffer.getInt();
```

Comment:
This one will be not int. First Short in the ByteBuffer will be MessageType (short).